synapseclient/client.py

```diff
@@ -1713,8 +1717,7 @@ def updateActivity(self, activity):
         :returns: An updated Activity object
         """
 
-        uri = '/activity/%s' % activity['id']
-        return Activity(data=self.restPUT(uri, json.dumps(activity)))
+        return self.saveActivity(activity)
```

Comment:
id check `activity['id']` so that `updateActivity` wont allow accidental creation of a new activity on accident.